### PR TITLE
close #5342 add testcase

### LINKED ISSUE #5342: Enable manual destructor invocation for implicitly generated destructors
Right now, it does not appear to be possible to manually invoke the destructor for a type when that destructor is defined implicitly, ie, it contains a destructable type.

```nim
{.experimental.}

type
  A = object
  B = object
    a: A

proc `=destroy`(a: A) = echo "~A"

`=destroy`(A()) # Compiles and works as expected
# `=destroy`(B()) # Does not compile.
```

### PR DESCRIPTION
**Pull request by ringabout**
close #5342

### CODE REVIEW

File: tests/destructor/t5342.nim [hunk 1]
```diff
@@ -0,0 +1,22 @@
+discard """
+  matrix: "--gc:refc; --gc:arc"
+  output: '''
+~A
+~A
+~A
+~A
+'''
+"""
+
```
Comment (timotheecour) [inline tests/destructor/t5342.nim:11]:
```nim
type
  A = object
    id: int
  B = object
    a: A
proc `=destroy`(a: var A) = echo a.id
var x = A(id: 1)
var y = B(a: A(id: 2))
`=destroy`(x)
`=destroy`(y)
echo "here"
```

printing:
1
2
here
2
1

seems like a more strict test
(or you can have both (in same file), but I think this covers the old test well enough)

Comment (ringabout) [reply tests/destructor/t5342.nim:11]: thanks

File: tests/destructor/t5342.nim [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  output: '''
+1
+2
+here
+2
+1
+'''
+"""
+
+
```
Comment (timotheecour) [inline tests/destructor/t5342.nim:12]:
```nim
when true:
  type
    A = object
      id: int
    B = object
      a: A
  proc `=destroy`(a: var A) = echo a.id
  template main =
    var x = A(id: 1)
    var y = B(a: A(id: 2))
    `=destroy`(x)
    `=destroy`(y)
    echo "here"
  static: main()
  main()
```

* fails in VM
* fails in RT js

is that tracked somewhere or is that a new bug? if a new bug, maybe 5342 could be kept open and repurposed to specify js/vm

Comment (ringabout) [reply tests/destructor/t5342.nim:12]: Open a new issue? Is it implemented for VM or JS?

**Review by timotheecour — APPROVED** [review]

File: tests/destructor/t5342.nim (hunk 2 above)
Comment (timotheecour) [reply tests/destructor/t5342.nim:12]: => filed https://github.com/nim-lang/Nim/issues/17237

**Pull request merged**